database/engine/journalfile.c

```diff
@@ -340,29 +366,40 @@ static void restore_extent_metadata(struct rrdengine_instance *ctx, struct rrden
             uv_rwlock_wrlock(&pg_cache->metrics_index.lock);
             PValue = JudyHSIns(&pg_cache->metrics_index.JudyHS_array, temp_id, sizeof(uuid_t), PJE0);
             fatal_assert(NULL == *PValue); /* TODO: figure out concurrency model */
-            *PValue = page_index = create_page_index(temp_id);
+            *PValue = page_index = create_page_index(temp_id, ctx);
             page_index->prev = pg_cache->metrics_index.last_page_index;
             pg_cache->metrics_index.last_page_index = page_index;
             uv_rwlock_wrunlock(&pg_cache->metrics_index.lock);
         }
 
         descr = pg_cache_create_descr();
         descr->page_length = jf_metric_data->descr[i].page_length;
-        descr->start_time = start_time;
-        descr->end_time = end_time;
+        descr->start_time_ut = start_time_ut;
+        descr->end_time_ut = end_time_ut;
+        descr->update_every_s = (update_every_s > 0) ? (uint32_t)update_every_s : (page_index->latest_update_every_s);
```

Comment:
I ran coverity scan again on this branch and now we have this:

```sh
*** CID 358453:  High impact quality  (Y2K38_SAFETY)
/database/engine/journalfile.c: 379 in restore_extent_metadata()
373             }
374     
375             descr = pg_cache_create_descr();
376             descr->page_length = jf_metric_data->descr[i].page_length;
377             descr->start_time_ut = start_time_ut;
378             descr->end_time_ut = end_time_ut;
>>>     CID 358453:  High impact quality  (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "update_every_s" is cast to "uint32_t".
379             descr->update_every_s = (update_every_s > 0) ? (uint32_t)update_every_s : (page_index->latest_update_every_s);
380             descr->id = &page_index->id;
381             descr->extent = extent;
382             descr->type = page_type;
383             extent->pages[valid_pages++] = descr;
384             pg_cache_insert(ctx, page_index, descr);
```

Reply:
We can ignore this. It just stores the "update_every". Eventually we will need to deal with all the `Y2K38_SAFETY` warnings